ecosystem/typescript/sdk_v2/src/types/hex.ts

```diff
@@ -0,0 +1,91 @@
+import { bytesToHex, hexToBytes } from "@noble/hashes/utils";
+import { HexInput } from ".";
+
+/**
+ * NOTE: Do not use this class when working with account addresses, use AccountAddress.
+ *
+ * NOTE: When accepting hex data as input to a function, prefer to accept HexInput and
+ * then use the static helper methods of this class to convert it into the desired
+ * format. This enables the greatest flexibility for the developer.
+ *
+ * Hex is a helper class for working with hex data. Hex data, when represented as a
+ * string, generally looks like this, for example: 0xaabbcc, 45cd32, etc.
+ *
+ * You might use this class like this:
+ *
+ * ```ts
+ * getTransactionByHash(txnHash: HexInput): Promise<Transaction> {
+ *   const txnHashString = Hex.hexInputToString({ hexInput: txnHash });
+ *   return await getTransactionByHashInner(txnHashString);
+ * }
+ * ```
+ *
+ * This call to `Hex.hexInputToString` converts the HexInput to a hex string with a
+ * leading 0x prefix, regardless of what the input format was.
+ */
+export class Hex {
+  private data: Uint8Array;
+
+  constructor(args: { data: Uint8Array }) {
+    this.data = args.data;
+  }
+
+  static fromString(args: { str: string }): Hex {
+    let input = args.str;
+
+    if (input.startsWith("0x")) {
+      input = input.slice(2);
+    }
+
+    if (input.length === 0) {
+      throw new Error("Hex string is too short, must be at least 1 char long, excluding the optional leading 0x.");
+    }
+
+    return new Hex({ data: hexToBytes(input) });
+  }
+
+  static fromHexInput(args: { hexInput: HexInput }): Hex {
+    if (args.hexInput instanceof Uint8Array) return new Hex({ data: args.hexInput });
+    return Hex.fromString({ str: args.hexInput });
+  }
+
+  toUint8Array(): Uint8Array {
+    return this.data;
+  }
+
+  toStringWithoutPrefix(): string {
+    return bytesToHex(this.data);
+  }
+
+  toString(): string {
+    return `0x${this.toStringWithoutPrefix()}`;
+  }
+
+  static hexInputToBytes(args: { hexInput: HexInput }): Uint8Array {
+    const hex = Hex.fromHexInput(args);
+    return hex.toUint8Array();
+  }
+
+  static hexInputToString(args: { hexInput: HexInput }): string {
+    const hex = Hex.fromHexInput(args);
+    return hex.toString();
+  }
+
+  static hexInputToStringWithoutPrefix(args: { hexInput: HexInput }): string {
```

Comment:
is there a namespace conflict even if it's static? 

Would be nice if it's consistent, bc ultimately you'll call it as `Hex.toStringWithoutPrefix(...)`

Reply:
Overall this looks nice to work with, except for the single arg object 😅 For stuff like this, like simple conversion functions, it definitely seems annoying and unintuitive because you need to name two args even though it's a single input

Reply:
Yeah... But we agreed we'll take this for a spin, so let's see how it goes!

Reply:
I figure prefixing with `hexInput` makes it a bit clearer that it's taking a hex input and that it's a static function.